tidy3d/plugins/waveguide/rectangular_dielectric.py

```diff
@@ -210,41 +215,51 @@ def _set_array(cls, val):
             raise ValidationError("Values may not be negative.")
         return result
 
+    @pydantic.validator("clad_medium", always=True)
+    def _set_medium_list(cls, val):
```

Comment:
might be better as a `@property` because this could be confusing if the user supplies a single medium. Also could break existing scripts.
```py
def _medium_list(self) -> list[MediumType]:
    if isinstance(val, MediumType):
        return [self.clad_medium]
    return self.clad_medium
````

Reply:
The breaking problem would also happen for `clad_thickness` and `box_thickness` too. I'll set them up as properties as well.